lib/std/crypto/aes_cmac.zig

```diff
@@ -0,0 +1,171 @@
+/// RFC 4493 - The AES-CMAC Algorithm
+/// https://www.rfc-editor.org/rfc/rfc4493
+const std = @import("std");
+const Aes128 = std.crypto.core.aes.Aes128;
+const print = std.debug.print;
+
+pub fn AesCmac() type {
+    return struct {
+        const Self = @This();
+        var k1: [16]u8 = undefined;
+        var k2: [16]u8 = undefined;
+        var ctx: std.crypto.core.aes.AesEncryptCtx(Aes128) = undefined;
+
+        /// Create a new cmac context with the given key.
+        pub fn init(key: [16]u8) Self {
+            ctx = Aes128.initEnc(key);
+            k1 = [_]u8{0} ** 16;
+            k2 = [_]u8{0} ** 16;
+            ctx.encrypt(&k1, &k1);
+            makeKn(&k1, &k1);
+            makeKn(&k2, &k1);
+            return Self{};
+        }
+
+        /// Generates a cmac from the given message.
+        pub fn generate(self: Self, msg: []const u8) [16]u8 {
+            _ = self;
+
+            var flag: bool = undefined;
+            var Ml: [16]u8 = undefined;
+
+            // rounds
+            var n = (msg.len + 15) / 16;
```

Comment:
`std.math.divCeil()`

Reply:
If would have to be 'divTrunc', but that doesn't make sense since the dominator is a constant.

Reply:
That's rounding up to the next block, so that's exactly what `divCeil()` does. Nothing wrong with using it with a constant denominator.

Reply:
However, rounding up is wrong at this point.

Reply:
The number of blocks (not rounds!) is `divCeil(msg.len, 16)`, no?

Reply:
No, if the message has less than 16 bytes it must be padded accordingly. Then you have no round, see figure 2.3.